ansi/writer.go

```diff
@@ -12,18 +13,19 @@ type Writer struct {
 	ansiseq    bytes.Buffer
 	lastseq    bytes.Buffer
 	seqchanged bool
+	runeBuf    []byte
 }
 
 // Write is used to write content to the ANSI buffer.
 func (w *Writer) Write(b []byte) (int, error) {
-	for i, c := range b {
+	for _, c := range string(b) {
 		if c == '\x1B' {
 			// ANSI escape sequence
 			w.ansi = true
 			w.seqchanged = true
-			_ = w.ansiseq.WriteByte(c)
+			w.ansiseq.WriteRune(c)
 		} else if w.ansi {
-			_ = w.ansiseq.WriteByte(c)
+			w.ansiseq.WriteRune(c)
```

Comment:
It's probably fine to ignore the error here, but if we do so, we should omit the assignment as we did before, to declare our intent. It's a bit repetitive, but I wouldn't mind proper error handling here either.